app/src/main/java/com/example/securityapplication/SQLiteDBHelper.java

```diff
@@ -0,0 +1,141 @@
+                    COLUMN_AADHAR + " TEXT, " +
+                    COLUMN_LOCATION + " TEXT, " +
+                    COLUMN_IMEI + " TEXT " + ")";
+
+
+    /**Constructor*/
+
+    public SQLiteDBHelper(Context context) {
+        super(context, DB_name, null, DB_version);
+    }
+
+    @Override
+    public void onCreate(SQLiteDatabase sqLiteDatabase) {
+        sqLiteDatabase.execSQL(CREATE_TABLE_QUERY);
+    }
+
+    @Override
+    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {
+        sqLiteDatabase.execSQL("DROP TABLE IF EXISTS " + TABLE_NAME);
+        onCreate(sqLiteDatabase);
+    }
+
+    /**Adding user*/
+    public boolean addUser(User user){
+        SQLiteDatabase db = this.getWritableDatabase();
+        ContentValues contentValues = new ContentValues();
+
+        contentValues.put(COLUMN_NAME, user.getName());
+        contentValues.put(COLUMN_EMAIL, user.getEmail());
+        contentValues.put(COLUMN_PASSWORD, user.getPassword());
+        contentValues.put(COLUMN_GENDER, user.getGender());
+        contentValues.put(COLUMN_MOBILE,user.getMobile());
+        contentValues.put(COLUMN_AADHAR, user.getAadhar());
+        contentValues.put(COLUMN_LOCATION, user.getLocation());
+        contentValues.put(COLUMN_IMEI, user.getImei());
+
+        long result = db.insert(TABLE_NAME,null, contentValues);
+        db.close();
+        if (result == -1){
+            return false;
+        }
+        else{
+            return true;
+        }
+    }
+
+    /**Updating user*/
+    public void updateUser(User user){
+        SQLiteDatabase db = this.getWritableDatabase();
+        ContentValues contentValues = new ContentValues();
+
+        contentValues.put(COLUMN_NAME, user.getName());
+        contentValues.put(COLUMN_EMAIL, user.getEmail());
+        contentValues.put(COLUMN_PASSWORD, user.getPassword());
+        contentValues.put(COLUMN_GENDER, user.getGender());
+        contentValues.put(COLUMN_MOBILE,user.getMobile());
+        contentValues.put(COLUMN_AADHAR, user.getAadhar());
+        contentValues.put(COLUMN_LOCATION, user.getLocation());
+        contentValues.put(COLUMN_IMEI, user.getImei());
+
+        db.update(TABLE_NAME,contentValues,COLUMN_ID + "=?",
+                new String[]{String.valueOf(user.getId())});
+        db.close();
+    }
+
+    /**Checking if user is present*/
+    public boolean checkUser(String aadhar){
+        String[] columns = {
+                COLUMN_ID
+        };
+        SQLiteDatabase db = this.getReadableDatabase();
+        String selection = COLUMN_AADHAR + " = ?";
+        String[] selectionArgs = {aadhar};
+
+        Cursor cursor = db.query(TABLE_NAME,
+                columns,
+                selection,
+                selectionArgs,
+                null,
+                null,
+                null);
+        int cursorCount = cursor.getCount();
+        cursor.close();
+        db.close();
+        if(cursorCount > 0){
```

Comment:
Unique aadhar number checking logic is wrong. When aadhar is unique ,the count should be exactly equal to 1.  Or make aadhaar number a unique key